ert/ensemble_evaluator/builder/_job.py

```diff
@@ -35,42 +36,48 @@ class _UnixJob(_BaseJob):
     def __init__(  # pylint: disable=too-many-arguments
         self,
         id_: str,
+        index: str,
         name: str,
         step_source: str,
         executable: Path,
         args: Optional[Tuple[str, ...]] = None,
     ) -> None:
-        super().__init__(id_, name, step_source)
+        super().__init__(id_, index, name, step_source)
         self.executable = executable
         self.args = args
 
 
 class _FunctionJob(_BaseJob):
+    # pylint: disable=too-many-arguments
+    # index is required by job sorting for gui
     def __init__(
         self,
         id_: str,
+        index: str,
         name: str,
         step_source: str,
         command: bytes,
     ) -> None:
-        super().__init__(id_, name, step_source)
+        super().__init__(id_, index, name, step_source)
         self.command = command
 
 
 class _LegacyJob(_BaseJob):
     def __init__(
         self,
         id_: str,
+        index: str,
         name: str,
         ext_job: ExtJob,
     ) -> None:
-        super().__init__(id_, name, "")  # no step_source needed for legacy (pt.)
+        super().__init__(id_, index, name, "")  # no step_source needed for legacy (pt.)
         self.ext_job = ext_job
 
 
 class _BaseJobBuilder:
     def __init__(self) -> None:
         self._id: Optional[str] = None
+        self._index: Optional[str] = None
```

Comment:
Why did  you choose string? Just curious

Reply:
Good point. As it being converted to json and back I thought better to have it directly in `str`. But if you prefer int, I agree to change it.